prime-router/src/test/kotlin/fhirengine/translation/hl7/utils/FhirPathUtilsTests.kt

```diff
@@ -73,6 +73,20 @@ class FhirPathUtilsTests {
         assertThat(FhirPathUtils.evaluateCondition(null, bundle, bundle, bundle, "")).isFalse()
     }
 
+    @Test
+    fun `test evaluate condition error thrown`() {
+        val bundle = Bundle()
+        bundle.id = "abc123"
+
+        val path = "Bundle.entry[0].resource.blah('blah')"
+        try {
+            FhirPathUtils.evaluateCondition(null, bundle, bundle, bundle, path)
+        } catch (e: Exception) {
+            assertThat(e).isInstanceOf<SchemaException>()
+            assertThat(e.cause!!.message).isEqualTo("Error in ?? at 1, 1: The name blah is not a valid function name")
```

Comment:
You can chain assertions to have better output if they fail and update the type you are working with.

```kotlin
assertThat(e.cause). // nullable exception
  .isNotNull() // non-nullable exception
  .isInstanceOf<UnderlyingExceptionType>() // checking the type matches what we expect
```

Reply:
Cool idea, thanks!